### PR TITLE
Feat/chunk data upload

### PR DESCRIPTION
**Pull request by abgeorge7**
Description about what this pull request does.

Chunking sheepdog requests for data upload.

### New Features
- Implemented XXX

### Improvements
- Can handle more requests for sheepdog

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
Jenkins Build 2 : time taken 1 min 1 sec

**Comment by PlanXCyborg** [comment]
Jenkins Build 1 : time taken 1 min 1 sec

**Comment by PlanXCyborg** [comment]
Jenkins Build 4 : time taken 22 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 5 : time taken 36 min

File: src/Submission/MapDataModel.jsx [hunk 1]
```diff
@@ -138,21 +141,38 @@ class MapDataModel extends React.Component {
         submitter_id: this.state.parentNodeId,
       };
       json.push(obj);
+
+      if (json.length === 10) {
```
Comment (qingyashu) [inline src/Submission/MapDataModel.jsx]: Is this always going to be `10`? How about let's make this a constant?

File: src/Submission/ReduxMapFiles.js [hunk 2]
```diff
@@ -3,16 +3,22 @@ import MapFiles from './MapFiles';
...
+const fetchUnmappedFiles = (user, total, start) => dispatch => fetchWithCreds({
+  path: `${indexdPath}index?acl=null&uploader=${user}&start=${start}&limit=${FETCH_LIMIT}`,
   method: 'GET',
 }).then(
   ({ status, data }) => {
     switch (status) {
     case 200:
+      total = total.concat(data.records);
+      if (data.records.length === FETCH_LIMIT) {
+        return dispatch(fetchUnmappedFiles(user, total, data.records[FETCH_LIMIT - 1].did));
```
Comment (qingyashu) [inline src/Submission/ReduxMapFiles.js]: Not sure, I might be wrong but will this last record `data.records[FETCH_LIMIT - 1]` download from backend in the next fetch? Or there's some other code already remove the duplicated records...?

File: src/Submission/ReduxSubmissionHeader.js [hunk 3]
```diff
@@ -25,7 +32,25 @@ const fetchUnmappedFileStats = user => dispatch => fetchWithCreds({
...
+}).then(
+  ({ status, data }) => {
+    switch (status) {
+    case 200:
+      return data.records[0].did;
+    default:
+      return null;
+    }
+  },
+).then(res => dispatch(fetchUnmappedFileStats(user, [], res)));
```
Comment (qingyashu) [inline src/Submission/ReduxSubmissionHeader.js]: I noticed that this `getStartingUUID` and `fetchUnmappedFileStats` are similar to previous `getStartingUUID` and `fetchUnmappedFiles` in `ReduxMapFiles.js`, maybe some common code could moved into util files. But this is not urgent to this PR if we just want a quick fix MVP

File: src/Submission/ReduxMapFiles.js [hunk 4]
```diff
@@ -22,7 +28,25 @@ const fetchUnmappedFiles = user => dispatch => fetchWithCreds({
...
+});
+
+const getStartingUUID = user => dispatch => fetchWithCreds({
+  path: `${indexdPath}index?acl=null&uploader=${user}&limit=1`,
+  method: 'GET',
+}).then(
+  ({ status, data }) => {
+    switch (status) {
+    case 200:
+      return data.records[0].did;
```
Comment (qingyashu) [inline src/Submission/ReduxMapFiles.js]: What if here `records` is empty array?

File: src/Submission/ReduxSubmissionHeader.js (hunk 3 above)
Comment (abgeorge7) [reply src/Submission/ReduxSubmissionHeader.js]: I agree it's tech debt, I'll try to move it out to a common function

File: src/Submission/ReduxMapFiles.js (hunk 2 above)
Comment (abgeorge7) [reply src/Submission/ReduxMapFiles.js]: I'm not sure honestly... the way pagination works with IndexD is via DID. So to paginate you have to take the last file did and pass it in as a starting point. I'm assuming the starting point is exclusive?

Comment (qingyashu) [reply src/Submission/ReduxMapFiles.js]: 🤔then what about the very first record of the first fetch, will it also be excluded?

**Comment by PlanXCyborg** [comment]
Jenkins Build 7 : time taken 25 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 6 : time taken 36 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 8 : time taken 27 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 9 : time taken 27 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 10 : time taken 1 min 12 sec

**Comment by PlanXCyborg** [comment]
Jenkins Build 11 : time taken 32 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 12 : time taken 16 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 13 : time taken 58 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 15 : time taken 42 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 16 : time taken 49 min

**Comment by PlanXCyborg** [comment]
Jenkins Build 17 : time taken 39 min

**Pull request merged**